File: src/lib/api/ddl.ts
import { invoke } from "./client";
import type { DdlRequest } from "./types";

// Generates the engine SQL for a DDL request (shown in the preview modal). The
// generated SQL is executed through the normal run_query path.
export const ddlApi = {
  generate: (sessionId: string, request: DdlRequest) =>
    invoke<string>("ddl_generate", { sessionId, request }),
};

File: src-tauri/src/commands/ddl.rs
//! Thin DDL command: generate the SQL for a structured request. Execution goes
//! through the normal `run_query` path (which owns the gates and schema refresh).

use tauri::State;

use crate::drivers::types::DdlRequest;
use crate::services::ddl_service;
use crate::state::AppState;
use crate::AppResult;

/// Returns the engine-specific SQL for a DDL request, for the preview modal.
#[tauri::command]
pub async fn ddl_generate(
    session_id: String,
    request: DdlRequest,
    state: State<'_, AppState>,
) -> AppResult<String> {
    ddl_service::generate(&session_id, &request, &state.sessions).await
}
